src/py42/_internal/archive_access.py

```diff
@@ -98,22 +99,22 @@ def __init__(
         archive_session_id,
         storage_archive_client,
         restore_job_manager,
+        file_size_poller,
     ):
         self._device_guid = device_guid
         self._archive_session_id = archive_session_id
         self._storage_archive_client = storage_archive_client
         self._restore_job_manager = restore_job_manager
+        self._file_size_poller = file_size_poller
 
-    def stream_from_backup(self, file_paths, ignore_size_calc=False):
+    def stream_from_backup(self, file_paths, file_size_calc_timeout=None):
         if not isinstance(file_paths, (list, tuple)):
             file_paths = [file_paths]
-
+        file_paths = [fp.replace(u"\\", u"/") for fp in file_paths]
         metadata_list = self._get_restore_metadata(file_paths)
         file_ids = [md[u"id"] for md in metadata_list]
-        file_sizes = (
-            self._get_file_size_info(file_ids)
-            if not ignore_size_calc
-            else _get_default_file_size_info(file_ids)
+        file_sizes = self._file_size_poller.get_file_sizes(
```

Comment:
Are the calculated file sizes accessible by the user at any point in the process of creating/streaming a restore job? I wonder if it would be useful to somehow expose the job size, so if it's very large there will be some expectation that it'll take longer. Maybe at least log the results at debug level? The file size polling could probably also use some progress logging now that I'm thinking of it.

Reply:
Interesting. I think debug works now, but here is an idea for how we can achieve this:

Create a new logger specifically for Restores (and size polling)... since python loggers are singletons, we can allow this logger to be exportable. The end user would then add a handler that sends the logs to wherever, such as updating UIs.

Reply:
I added the debugging logging and tested it out. It works! I have it like JSON so it looks like the other logging we do. Also, File size calculation does not take nearly as long as the actual restore.

Reply:
Like, the restore is magnitudes longer... so perhaps that is where the real progress bar should happen. I restored my entire C drive (which was still pretty small), and I did the size calculation in "WORKING" state for a very short amount of time.

Reply:
Should we log progress during restore?

Reply:
Yeah, most calculations happen pretty quick, but if you've got an archive with a huge number of files, it can spin for quite awhile, so it would be useful in those cases to be able to see that it's doing _something_ and whether it is making progress or not. 

Yeah, we should log anytime there's an operation happening behind the scenes that the user might want to know about.

Reply:
Well, we are logging requests. So, we can tell that it _is_ doing something. This was true for sile polling too. But logging progress is just the next step.

Reply:
Ok, logging percent complete now! I agree this is much better for the logs